File: thor/src/jvmMain/kotlin/io/github/remmerw/thor/cobra/SimpleUserAgentContext.java
package io.github.remmerw.thor.cobra;

import java.net.URL;

import io.github.remmerw.thor.cobra.ua.NetworkRequest;
import io.github.remmerw.thor.cobra.ua.UserAgentContext;

public class SimpleUserAgentContext implements UserAgentContext {
    @Override
    public boolean isRequestPermitted(Request request) {
        return false;
    }

    @Override
    public NetworkRequest createHttpRequest() {
        return null;
    }

    @Override
    public String getAppCodeName() {
        return "";
    }

    @Override
    public String getAppName() {
        return "";
    }

    @Override
    public String getAppVersion() {
        return "";
    }

    @Override
    public String getAppMinorVersion() {
        return "";
    }

    @Override
    public String getBrowserLanguage() {
        return "";
    }

    @Override
    public boolean isCookieEnabled() {
        return false;
    }

    @Override
    public boolean isScriptingEnabled() {
        return false;
    }

    @Override
    public boolean isExternalCSSEnabled() {
        return false;
    }

    @Override
    public boolean isInternalCSSEnabled() {
        return false;
    }

    @Override
    public String getPlatform() {
        return "";
    }

    @Override
    public String getUserAgent() {
        return "";
    }

    @Override
    public String getCookie(URL url) {
        return "";
    }

    @Override
    public void setCookie(URL url, String cookieSpec) {

    }

    @Override
    public int getScriptingOptimizationLevel() {
        return 0;
    }

    @Override
    public boolean isMedia(String mediaName) {
        return false;
    }

    @Override
    public String getVendor() {
        return "";
    }

    @Override
    public String getProduct() {
        return "";
    }
}

File: thor/src/jvmMain/kotlin/io/github/remmerw/thor/cobra/SimpleUserAgentContext.kt
package io.github.remmerw.thor.cobra

import io.github.remmerw.thor.cobra.ua.NetworkRequest
import io.github.remmerw.thor.cobra.ua.UserAgentContext
import java.net.URL

class SimpleUserAgentContext : UserAgentContext {
    override fun isRequestPermitted(request: UserAgentContext.Request?): Boolean {
        return false
    }

    override fun createHttpRequest(): NetworkRequest? {
        return null
    }

    override fun getAppCodeName(): String {
        return ""
    }

    override fun getAppName(): String {
        return ""
    }

    override fun getAppVersion(): String {
        return ""
    }

    override fun getAppMinorVersion(): String {
        return ""
    }

    override fun getBrowserLanguage(): String {
        return ""
    }

    override fun isCookieEnabled(): Boolean {
        return false
    }

    override fun isScriptingEnabled(): Boolean {
        return false
    }

    override fun isExternalCSSEnabled(): Boolean {
        return true
    }

    override fun isInternalCSSEnabled(): Boolean {
        return true
    }

    override fun getPlatform(): String {
        return ""
    }

    override fun getUserAgent(): String {
        return ""
    }

    override fun getCookie(url: URL?): String {
        return ""
    }

    override fun setCookie(url: URL?, cookieSpec: String?) {
    }

    override fun getScriptingOptimizationLevel(): Int {
        return 0
    }

    override fun isMedia(mediaName: String?): Boolean {
        return false
    }

    override fun getVendor(): String {
        return ""
    }

    override fun getProduct(): String {
        return ""
    }
}
